lib/AST/ASTImporter.cpp

```diff
@@ -3194,7 +3213,22 @@ Decl *ASTNodeImporter::VisitVarDecl(VarDecl *D) {
             D->hasExternalFormalLinkage()) {
           if (Importer.IsStructurallyEquivalent(D->getType(),
                                                 FoundVar->getType())) {
-            MergeWithVar = FoundVar;
+
+            // The VarDecl in the "From" context has a definition, but in the
+            // "To" context we already has a definition.
+            VarDecl *FoundDef = FoundVar->getDefinition();
+            if (D->isThisDeclarationADefinition() && FoundDef) {
+              return Importer.MapImported(D, FoundDef);
+            }
```

Comment:
Is there no problem with update of initializer? If the `D` is definition and has initializer, and the found has no initializer, it is not updated. (But theoretically the variable can have only one definition, so the definition of `D` should be the same as the found one, with initializer.)

Reply:
Hmm, good point, but I don't think that could be a problem exactly because what you have written in the parentheses.
I would not merge the initializer intstead, we could diagnose an ODR error here if we see that the two definitions have different initializers, I added a FIXME for that.